### PR TITLE
Database Refactor

### PR DESCRIPTION
**Pull request by cmbrose**
## What

A restructuring of how we access databases:
- Most of the `postgres` package moved to a new `databases` package
- Instead of many database connections each supporting one table, one connection with many tables
## Why

This will help us add support for future (SQL based) databases
## How
- `postgres` simply offers a singleton database connection
- `databases` takes a table name and a database connection (the one given by `postgres` for example) and works its magic

### CODE REVIEW

**Comment by cmbrose** [comment]
@lighthouse/owners

File: databases/databases.go [hunk 1]
```diff
@@ -1,167 +1,117 @@
...
+}
+
+func (this *Table) Update(key string, newValue interface{}) (error) {
+    json, _ := json.Marshal(newValue)
+    query := fmt.Sprintf(`UPDATE %s SET %s = ($2) WHERE %s = ($1);`,
+        this.table, valueColumn, keyColumn)
+
+    res, err := this.db.Exec(query, key, string(json))
+
+    fmt.Printf("%s\n", res)
```
Comment (robertsheehy-wf) [inline databases/databases.go]: Is this still necessary? If so, can we make it a `logging.Info` statement.

**Comment by robertsheehy-wf** [comment]
+1

**Comment by nickmonad** [comment]
+1

**Comment by chfogerty** [comment]
+1

**Pull request merged**